modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTests.java

```diff
@@ -426,6 +426,55 @@ void deleteOldChunks(String name, int firstChunk) {
         assertEquals(0, stats.getFailedDownloads());
     }
 
+    public void testCleanDatabases() throws IOException {
+        ByteArrayInputStream bais = new ByteArrayInputStream(new byte[0]);
+        when(httpClient.get("http://a.b/t1")).thenReturn(bais);
+
+        final AtomicInteger count = new AtomicInteger(0);
+
+        geoIpDownloader = new GeoIpDownloader(
+            client,
+            httpClient,
+            clusterService,
+            threadPool,
+            Settings.EMPTY,
+            1,
+            "",
+            "",
+            "",
+            EMPTY_TASK_ID,
+            Map.of(),
+            () -> GeoIpDownloaderTaskExecutor.POLL_INTERVAL_SETTING.getDefault(Settings.EMPTY),
+            () -> GeoIpDownloaderTaskExecutor.EAGER_DOWNLOAD_SETTING.getDefault(Settings.EMPTY),
+            () -> true
+        ) {
+            @Override
+            void updateDatabases() throws IOException {
+                // noop
+            }
+
+            @Override
+            void deleteOldChunks(String name, int firstChunk) {
+                count.incrementAndGet();
+                assertEquals("test.mmdb", name);
+                assertEquals(21, firstChunk);
+            }
+
+            @Override
+            void updateTaskState() {
+                // noop
+            }
+        };
+
+        geoIpDownloader.setState(GeoIpTaskState.EMPTY.put("test.mmdb", new GeoIpTaskState.Metadata(10, 10, 20, "md5", 20)));
+        geoIpDownloader.runDownloader();
+        geoIpDownloader.runDownloader();
+        GeoIpDownloaderStats stats = geoIpDownloader.getStatus();
+        assertEquals(1, stats.getExpiredDatabases());
+        assertEquals(2, count.get()); // somewhat surprising, not necessarily wrong
+        assertEquals(18, geoIpDownloader.state.getDatabases().get("test.mmdb").lastCheck()); // highly surprising, seems wrong
```

Comment:
Can you explain why this is surprising and seems wrong? Does it need more investigation?

Reply:
@joegallo  and I *think* that this line was probably a mistake: https://github.com/elastic/elasticsearch/blob/main/modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java#L330. We can't think of a valid reason to subtract 1 from the last checked millis every time we run `cleanDatabases`. We suspect that the intention was to set it to `-1` as a marker for a future check. Right now it doesn't really do any harm, and it was outside of the scope of this PR. So we just tested the behavior as-is, with a note that we might want to come back to it at some point. There had been no code coverage on this behavior before that I could fine.